tools/scylla-sstable.cc

```diff
@@ -299,14 +318,257 @@ class dumping_consumer : public sstable_consumer {
             return make_ready_future<>();
         }
     };
+    class json_dumper : public sstable_consumer {
+        const schema& _schema;
+        json_writer _writer;
+        bool _clustering_array_created;
+    private:
+        sstring to_string(gc_clock::time_point tp) {
+            return fmt::format("{:%F %T}", fmt::gmtime(gc_clock::to_time_t(tp)));
+        }
+        void write(gc_clock::duration ttl, gc_clock::time_point expiry) {
+            _writer.Key("ttl");
+            _writer.AsString(ttl);
+            _writer.Key("expiry");
+            _writer.String(to_string(expiry));
+        }
+        template <typename Key>
+        void write_key(const Key& key) {
+            _writer.StartObject();
+            _writer.Key("raw");
+            _writer.String(to_hex(key.representation()));
+            _writer.Key("value");
+            _writer.AsString(key.with_schema(_schema));
+            _writer.EndObject();
+        }
+        void write(const tombstone& t) {
+            _writer.StartObject();
```

Comment:
Callers are expected to avoid calling this method if `!t`. If we are here, we already wrote a key, so we must write a value so the JSON stays well-formed. If `!t` we have no choice but to write an empty object.

Reply:
Oh, ok. Makes (a bit of) sense ;-)